plugin/trino-vertica/src/main/java/io/trino/plugin/vertica/VerticaClient.java

```diff
@@ -228,6 +235,8 @@ public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connect
                         DATE,
                         (resultSet, index) -> LocalDate.parse(resultSet.getString(index), DATE_READ_FORMATTER).toEpochDay(),
                         dateWriteFunctionUsingString()));
+            case Types.TIMESTAMP:
+                return Optional.of(timestampColumnMappingUsingString(createTimestampType(typeHandle.requiredDecimalDigits())));
```

Comment:
> Vertica jdbc driver doesn't differentiate this types and use java.sql.Timestamp

I think we can recognize the type with `JdbcTypeHandle#jdbcTypeName`. Vertica returns `Timestamp` for timestamp type and `TimestampTz` for timestamp with time zone.

Reply:
I think I need to rephrase this sentence, I mean - that Vertica jdbc driver use  java.sql.Timestamp for both of this types, despite other drivers use java.time.LocalDateTime for timestamp without Time Zone

Reply:
But we could get the value as String and insert the data as String right ?

Reply:
As far as I understand the code, it looks like this:
1) in `DefaultQueryBuilder` for predicate pushdown we use `prepareStatement` method, where we 
```
           else if (javaType == long.class) {
                ((LongWriteFunction) writeFunction).set(statement, parameterIndex, (long) value);
            }
```

so we took our write function for timestamp and set value for prepare statement.
2) Write function should implement something like this:
```
        return LongWriteFunction.of(Types.TIMESTAMP, (statement, index, value) -> {
            statement.setObject(index, "CAST(TIMESTAMP '%s')".formatted(FORMATTER_SIX_SCALE.format(fromTrinoTimestamp(value))));
        });
```

`or just  statement.setString('some date')`

3) But the problem is inside driver,
when driver see that column type is TIMESTAMP it tries to convert it to java.sql.Timestamp:
```
case 12:
                        value = (String)data.getObject();
                        boolean datePresent = value.contains("-");
                        boolean timePresent = value.contains(":");
                        if (datePresent && timePresent) {
                            return StringConverter.parseTimestamp(value, new GregorianCalendar(null == calendar ? TimeZone.getDefault() : calendar.getTimeZone()));
                        }
```

So whatever we set in prepared statement driver will try to convert it to java.sql.Timestamp

So the only way how we can overcome it - is to use statement, instead of prepareStatement. But not sure it's possible

Reply:
But we won't be setting it as a Timestamp right ? In case of Druid we try something like this 
```
        return new LongWriteFunction()
        {
            @Override
            public String getBindExpression()
            {
                return "CAST(? AS TIMESTAMP)";
            }

            @Override
            public void set(PreparedStatement statement, int index, long value)
                    throws SQLException
            {
                statement.setString(index, LOCAL_DATE_TIME.format(fromTrinoTimestamp(value)));
            }
        };
```

Here the object value which we are injecting would be a String - so it won't be converted to java.sql.Timestamp right ?

Reply:
No, looks like conversion is still happened :(

Reply:
I dived a little bit inside vertica jdbc driver and it looks like this, during creating PreparedStatement, driver performs call to Vertica DB, to collect information about data types:

```
var4 = this.m_statement.makeStatementName();
var6 = false;
int[] var38 = new int[var6];
Arrays.fill(var38, 0);
this.m_protocolStream.sendMessage(new ParseRequestMessage(var4, var1, var38));
this.m_protocolStream.sendMessage(new DescribeRequestMessage(false, var4));
this.m_protocolStream.sendMessage(new FlushRequestMessage());
this.m_protocolStream.readExpectedMessage(ParseRequestMessage.LEGAL_RESPONSE);
ParameterDescriptionResponseMessage var8 = (ParameterDescriptionResponseMessage)this.m_protocolStream.readExpectedMessage(DescribeRequestMessage.LEGAL_RESPONSE_STEP1);
int[] var9 = var8.getParameterTypes();
int[] var10 = var8.getParameterTypmods();
String[] var11 = var8.getParameterTypeNames();
boolean[] var12 = var8.getParameterNotNulls();
```

So before setting up args for PreparedStatement, it has understanding of types, which will be used in query.
So even if use `statement.setString `or `statement.setObject`
driver is able to perform conversion to java.sql.Timestamp:

so we cannot overcome this problem as in Druid connector.

However there is a place, where driver have such code:
```
if (paramMeta.shouldConvertInputToString()) {
                data = this.convertValueToTypeAsString(data, toTypeMetadata);
            } else {
                data = TypeConverter.toType(data, toTypeMetadata, this.m_warningListener);
            }
```

When I forcibly in debug mode set this flag to true, it doesn't perform java.sql.Timestamp conversion and behaves exactly as other drivers and all our test corner cases passes.

But I didn't find the way how to change this flag in normal way, outside the driver :(